File: src/test/samples/simple/src/main/java/io/vlingo/gradle/actortest/Test1ProtocolActor.java
package io.vlingo.gradle.actortest;

import java.util.List;

import io.vlingo.actors.Actor;

public class Test1ProtocolActor extends Actor implements Test1Protocol {
    public static int doSomethingElseWithValue;
    public static String doSomethingWith;

    @Override
    public void doSomethingWith(String name, List<String> texts) {
        final StringBuilder builder = new StringBuilder();

        builder.append(name);

        for (final String text : texts) {
            builder.append(text);
        }

        doSomethingWith = builder.toString();
    }

    @Override
    public void doSomethingElseWith(int value) {
        doSomethingElseWithValue = value;
    }
}

File: src/test/resources/protocols/main/scala/io/vlingo/gradle/actortest/Test1ProtocolActor.scala
package io.vlingo.gradle.actortest

import io.vlingo.actors.Actor

import scala.beans.BeanProperty
import scala.collection.JavaConverters._

class Test1ProtocolActor extends Actor with Test1Protocol {

    override def doSomethingWith(name: String, texts: java.util.List[String]): Unit = {
        val builder = new StringBuilder()
        builder.append(name);
        for (text <- texts.asScala) {
            builder.append(text)
        }
        Test1ProtocolActor.doSomethingWith = builder.toString()
    }

    override def doSomethingElseWith(value: Int): Unit = {
        Test1ProtocolActor.doSomethingElseWithValue = value;
    }
}

object Test1ProtocolActor {
    @BeanProperty var doSomethingElseWithValue: Int = -1
    @BeanProperty var doSomethingWith: String = null
}
